src/sage/combinat/combinat.py

```diff
@@ -2963,6 +2991,28 @@ def bell_polynomial(n: Integer, k: Integer):
         [[], [1], [3], [1]]
         [[], [1], [3, 4], [6], [1]]
 
+    Further checks for :issue:`37727`::
+
+        sage: # needs sage.combinat
+        sage: # needs sage.arith
+        sage: bell_polynomial(0, 0)
+        1
+        sage: bell_polynomial(0, 0, True)
+        1
+        sage: bell_polynomial(1, 1)
+        x0
+        sage: bell_polynomial(2, 2, True)
+        2*x0^2
+        sage: bell_polynomial(5)
+        x0^5 + 10*x0^3*x1 + 15*x0*x1^2 + 10*x0^2*x2 + 10*x1*x2 + 5*x0*x3 + x4
+        sage: sum(bell_polynomial(5, k) for k in (0..5))
+        x0^5 + 10*x0^3*x1 + 15*x0*x1^2 + 10*x0^2*x2 + 10*x1*x2 + 5*x0*x3 + x4
+        sage: bell_polynomial(5, None, True)
+        120*x0^5 + 60*x0^3*x1 + 30*x0*x1^2 + 20*x0^2*x2 + 10*x1*x2 + 5*x0*x3 + x4
+        sage: sum(bell_polynomial(5, k, True) for k in (0..5))
```

Comment:
```suggestion
        sage: sum(bell_polynomial(5, k) for k in range(5))
        x0^5 + 10*x0^3*x1 + 15*x0*x1^2 + 10*x0^2*x2 + 10*x1*x2 + 5*x0*x3 + x4
        sage: bell_polynomial(5, None, True)
        120*x0^5 + 60*x0^3*x1 + 30*x0*x1^2 + 20*x0^2*x2 + 10*x1*x2 + 5*x0*x3 + x4
        sage: sum(bell_polynomial(5, k, True) for k in range(5))
```
Better to use standard Python here.

Reply:
For this one it should be `range(6)` then. But I figured it was more readable with the sage notations.